tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1114 @@
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*5,
+		time.Minute*1,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Object.Name, clientPod.Definition.Namespace)
+
+			output = result.String()
+			glog.V(100).Infof("Command's output:\n\t%v", output)
+
+			return true, nil
+		})
+
+	if err != nil {
+		return "", fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+	}
+
+	return output, nil
+}
+
+func generateLongTCPTraffic(
+	clientPod *pod.Builder,
+	serverIPAddr,
+	serverPort,
+	packetsNumber,
+	timeout string) (string, error) {
+	By("Open long TCP session and generate traffic")
+
+	glog.V(100).Infof("Ensure pod %q in namespace %q is Ready",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	err := clientPod.WaitUntilReady(5 * time.Second)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to wait for pod %q in namespace %q to become Ready: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return "", fmt.Errorf("failed to wait for pod %q in namespace %q to become Ready: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -packages %s -timeoutTCP %s -server %s",
+			serverPort, packetsNumber, timeout, serverIPAddr)}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*3,
+		time.Minute,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Object.Name, clientPod.Definition.Namespace)
+
+			output = result.String()
+			glog.V(100).Infof("Command's output:\n\t%v", output)
+
+			return true, nil
+		})
+
+	if err != nil {
+		return "", fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+	}
+
+	return output, nil
+}
+
+func findInCmdExecOutput(cmdExecOutput, stringToFind string) (bool, error) {
+	var err error
+
+	matchesFound := 0
+
+	if len(cmdExecOutput) != 0 {
+		buf := new(bytes.Buffer)
+		_, err = buf.WriteString(cmdExecOutput)
+
+		stringToFind, _ := regexp.Compile(stringToFind)
```

Comment:
```suggestion
		stringToFind, err := regexp.Compile(stringToFind)
```

Reply:
if there's a check for writing to buffer(based on the following messages) then this `regexp.Compile` call should be done after error's evaluation